ego/cli/install.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	"github.com/klauspost/cpuid/v2"
+	"github.com/zcalusic/sysinfo"
+)
+
+const shellToUse = "bash"
+
+var ErrOSNotSupported = errors.New("os not supported")
+var ErrTargetNotSupported = errors.New("install target not supported")
+var ErrInstallationFail = errors.New("installation fail")
+var ErrExitCodeValue = errors.New("exit code not 0")
+var ErrSysInfoFail = errors.New("could not determine os details")
+
+type installInfoV1 struct {
+	Desc     map[string]string
+	Commands map[string]map[string][]string
+}
+
+type installInfo struct {
+	V1 installInfoV1 `json:"1.0"`
+}
+
+func (i installInfo) getDescriptions() map[string]string {
+	return i.V1.Desc
+}
+
+func (i installInfo) getCommands(osDetails string) map[string][]string {
+	return i.V1.Commands[osDetails]
+}
+
+// Determine relevant information about os and sgx compatibility. Then pass them with osDetails to install
+func (c *Cli) Install(ask func(string) bool, component string) error {
+
+	var si sysinfo.SysInfo
+	si.GetSysInfo()
+	vendor := si.OS.Vendor
+	version := si.OS.Version
+	if vendor == "" || version == "" {
+		fmt.Println("Could not determine necessary details about operating system. Exiting...")
```

Comment:
```suggestion
```
Remove all the error messages from this function and let the caller do this instead based on the returned error. (separate UI and logic)